assets/quick-order-list.js

```diff
@@ -113,7 +113,10 @@ if (!customElements.get('quick-order-list')) {
 
       connectedCallback() {
         this.cartUpdateUnsubscriber = subscribe(PUB_SUB_EVENTS.cartUpdate, (event) => {
-          if (event.source === this.quickOrderListId) {
+          if (
+            event.source === this.quickOrderListId ||
+            (event.cartData.items && event.cartData.items.find((item) => item.id === this.dataset.id) === undefined)
```

Comment:
`dataset.id` appears to be the section ID, not a variant ID. So this will never find a match. Doesn't appear that there is variant information available here since its at the level of the entire list.

Outputting `this.dataset`: 

![](https://screenshot.click/19-44-hsx40-fsb0l.png)

Reply:
good point 👍 https://github.com/Shopify/dawn/pull/3480/commits/28b33ed923be6a174e380fee7c217842452fbcd1